test/integration/lib/philipsHue_server.js

```diff
@@ -0,0 +1,83 @@
+var express = require ('express');
+var bodyParser = require('body-parser');
+var path = require('path');
+
+var philipsHue_server = (function() {
+
+	var welcome_page = '/html/philips_initial.html';		// Philips Hub webpage
+	var suc_resp = require('./success_response.js');		// TBD: move this to appropriate module
+	var api_resp = require('./json/bridge_status.json');	// full status response
+	var light_status_resp = require('./json/lights_status.json'); // light status response
+	
+	var commanded_light_state ={1:true,2:true,3:true};		// TBD: move this to appropriate module
+	var last_received_cmd = null;		
+
+	return {
+
+		setlight: function(id,value) {  // Set the on/off status of simulated light
+			commanded_light_state[id] = value;
+		},
+
+		get last_cmd() {	
+			return last_received_cmd;
+		},
+
+		setup : function(port) {
+			var _server = express();
+			var foxbox_id = 'none';
+
+
+			_server.use(function (req, res, next) {	// foxbox does not have the json header, need to inject it here so the message boy
+													// can be properly captured
+			    req.headers['content-type'] = 'application/json';
+			    last_received_cmd = req.method + ":" + req.originalUrl;
+				next();
+			});
+			
+			_server.get('/', function (req, res) {
+				res.status(200).sendFile(path.join(__dirname + welcome_page));  
+			});
+
+
+			_server.get('/api/:foxbox/', function (req, res) {   
+				if (foxbox_id == 'none') {	// foxbox id is created within foxbox.  Need to remember this
+					foxbox_id = req.params.foxbox;
+					res.sendStatus(200); 
+				}
+				else {
+					res.status(200).json(api_resp);  
+				} 		
+			});
+
+			_server.get('/api/:foxbox_id/lights', function (req, res) {
+				if (req.params.foxbox_id == foxbox_id){
+					res.status(200).json(light_status_resp); 
+				}
```

Comment:
* No need to add it in this PR *
Improvement: Make the server return a failure if the foxbox_id is not the expected one.